firebase-appdistribution/src/test/java/com/google/firebase/appdistribution/ApkUpdaterTest.java

```diff
@@ -238,4 +248,37 @@ public void updateApp_whenCalledMultipleTimesWithApk_returnsSameUpdateTask() {
 
     assertThat(updateTask1).isEqualTo(updateTask2);
   }
+
+  @Test
+  public void updateApk_whenSuccessfullyUpdated_notificationsSetCorrectly()
+      throws FirebaseAppDistributionException {
+    doReturn(TEST_FILE_LENGTH)
+        .when(apkUpdater)
+        .downloadToDisk(any(), anyLong(), anyString(), anyBoolean());
+    doNothing().when(apkUpdater).validateJarFile(any(), anyLong(), anyBoolean(), anyLong());
+
+    UpdateTask updateTask = apkUpdater.updateApk(TEST_RELEASE, true);
+    updateTask.addOnCompleteListener(testExecutor, onCompleteListener);
+    List<UpdateProgress> progressEvents = new ArrayList<>();
+    updateTask.addOnProgressListener(testExecutor, progressEvents::add);
+
+    assertThat(progressEvents).hasSize(2);
+    assertThat(progressEvents.get(0).getUpdateStatus()).isEqualTo(UpdateStatus.PENDING);
+    assertThat(progressEvents.get(1).getUpdateStatus()).isEqualTo(UpdateStatus.DOWNLOADED);
+  }
+
+  @Test
+  public void downloadToDisk_whenHasInputStream_notificationSetCorrectlyHasCorrectInputLength()
```

Comment:
Added two tests.

 One to check the notifications for makeApkDownloadRequest through updateApk. 

Another to test the read and writing for the downloadToDisk method. And one on the negative path making sure the notification set.